webapp/VersionSummary.py

```diff
@@ -35,8 +37,14 @@ def aggregate_descriptors(descriptors, descriptor_filter, version_to_tag):
 
         version_summaries[descriptor.version].update(descriptor)
 
+    for node_descriptor in node_descriptors:
+        if node_descriptor_filter and not node_descriptor_filter(node_descriptor):
+            continue
+
```

Comment:
You'll need to add a filter here to account for when a version is present in the nodes file, but not in the harvesters file. Something like:

```
if node_descriptor.version not in version_summaries:
            version_summaries[node_descriptor.version] = VersionSummary(version_to_tag(node_descriptor.version))
```